target/linux/realtek/dts-5.10/rtl8393_tplink_sg2452p-v4.dts

```diff
@@ -0,0 +1,330 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "rtl839x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "tplink,sg2452p-v4", "realtek,rtl838x-soc";
+	model = "TP-Link SG2452P v4";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x10000000>;
+	};
+
+	aliases {
+		led-boot = &led_sys;
+		led-failsafe = &led_sys;
+		led-running = &led_sys;
+		led-upgrade = &led_sys;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,38400";
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
```

Comment:
The gpio driver of Realtek SoCs supports interrupt, so this can be replaced by ```gpio-keys```. Then, you can drop ```poll-interval``` property.

Reply:
OK, I didn't know that. My reference for this was the GS1900-48 where `gpio-keys-polled` is used, but now I saw that the Panasonic uses `gpio-keys`.

Reply:
GS1900-48 uses the pin on RTL8231 for keys and it doesn't support interrupt. Panasonic Switch-M48eG PN28480K uses the pin on PCA9539 controller, it supports interrupt via SoC GPIO controller.

Reply:
Oh, well, I should have had a closer look :) Thanks for the clarification!